src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Resolvers/DynamicObjectResolver.cs

```diff
@@ -873,6 +873,24 @@ private static void BuildDeserialize(Type type, ObjectSerializationInfo info, IL
 
             il.MarkLabel(falseLabel);
 
+            var canOverwrite = info.ConstructorParameters.Length == 0;
```

Comment:
This `canOverwrite` stuff is puzzling. You mentioned in the description I think that it was forward looking to supporting circular references. AFAIK we haven't settled on a design for it so I'm concerned about introducing complexity here before it's necessary. Do we really need it here as part of _this_ feature?
If we need to keep this, please add a code comment explaining what `canOverwrite` means and how constructor parameter length determines whether something can be overwritten.

Reply:
`canOverwrite` means `T where T : new()`.
If T can be instantiated without any parameter, you can instantiate it before field/property deserializations and assign deserialized values directly to its members. This is very performant.

```csharp
[MessagePackObject]
public struct float4
{
    [Key(0)] public float X;
    [Key(1)] public float Y;
    [Key(2)] public float Z;
    [Key(3)] public float W;
}

[MessagePackObject, StructLayout(LayoutKind.Explicit)]
public struct float4x4
{
    [FieldOffset(0), IgnoreMember] public Matrix4x4 Matrix;
    [FieldOffset(0), Key(0)] public float4 C0;
    [FieldOffset(16), Key(1)] public float4 C1;
    [FieldOffset(32), Key(2)] public float4 C2;
    [FieldOffset(48), Key(3)] public float4 C3;
}
```

The above `float4x4` is often defined by me.
The large size struct field/property overwriting is useful in such a case.

Reply:
Great. I thought `overwrite` had to do with overwriting an existing instance (related to another issue we have/had on this project). That makes much more sense. Thanks.